controllers/release/adapter.go

```diff
@@ -751,20 +912,32 @@ func (a *adapter) validatePipelineSource() *controller.ValidationResult {
 		return a.validationError(err)
 	}
 
-	if releasePlan.Spec.Pipeline != nil {
-		pipelineRef = releasePlan.Spec.Pipeline.PipelineRef
-	} else {
+	if releasePlan.Spec.TenantPipeline != nil {
+		pipelineRef = releasePlan.Spec.TenantPipeline.PipelineRef
+		if !a.releaseServiceConfig.Spec.Debug && pipelineRef.IsClusterScoped() {
+			a.release.MarkValidationFailed("tried using debug only options while debug mode is disabled in the ReleaseServiceConfig")
+			return &controller.ValidationResult{Valid: false}
+		}
+	}
+
+	if releasePlan.Spec.FinalPipeline != nil {
+		pipelineRef = releasePlan.Spec.FinalPipeline.PipelineRef
+		if !a.releaseServiceConfig.Spec.Debug && pipelineRef.IsClusterScoped() {
+			a.release.MarkValidationFailed("tried using debug only options while debug mode is disabled in the ReleaseServiceConfig")
+			return &controller.ValidationResult{Valid: false}
+		}
+	}
+
+	if releasePlan.Spec.TenantPipeline == nil {
```

Comment:
Confused why you added this. Using a tenant pipeline doesn't mean you can use any managed pipeline you want. Whether or not the tenant pipeline is nil has no bearing on the validation of the managed pipeline. If a managed pipeline is passed, it should be validated

Reply:
I was writing the comment :)
So, the reason for the change is when you run a tenant pipeline, only the function will still be checking for a RPA, causing an error. Previously, we had the if-else which would only check one or the other .

Reply:
Yeah that would make more sense

Reply:
That works updated it